kivy/core/image/__init__.py

```diff
@@ -677,7 +738,14 @@ def _img_iterate(self, *largs):
         imgcount = len(self.image.textures)
         if imgcount > 1:
             self._anim_available = True
-            self.anim_reset(True)
+
+            if (self._auto_anim_delay
```

Comment:
I think this should be more like:

```py
if self._auto_anim_delay:
    if imgcount != len(self.image.durations):
        raise ValueError(...)
    self.durations = self.image.durations
else:
...
```
Because we shouldn't quietly move on if there's an error and not matching lengths is an error (I believe)!? Similarly, do we need to check whether it's a float/int? If it isn't an error will be raised when you try to use it.

If you really want to check for an error, maybe above instead of doing `self._durations = list(durations)` like I suggested, simply do `self._durations = list(map(float, durations))`. This will convert everything floats and raise an error immediately if it can't. Then you don't have to do any checking here.